src/main/java/com/mixpanel/android/mpmetrics/InAppButton.java

```diff
@@ -0,0 +1,97 @@
+package com.mixpanel.android.mpmetrics;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class InAppButton implements Parcelable {
+
+    private static final String LOGTAG = "MixpanelAPI.InAppButton";
+
+    private JSONObject mDescription;
+    private String mText;
+    private int mTextColor;
+    private int mBackgroundColor;
+    private int mBorderColor;
+    private String mCtaUrl;
+
+    public InAppButton(Parcel in) {
+        JSONObject temp = new JSONObject();
```

Comment:
Definitely not the first one. The second one would work but I do not like it as much. Say that a method/constructor throws multiple exceptions; you'd need to make sure to init the variable you are trying to initialize in every single catch statement.